src/Avalonia.Controls/WindowClosingEventArgs.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.ComponentModel;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Specifies the reason that a window was closed.
+    /// </summary>
+    public enum WindowCloseReason
+    {
+        /// <summary>
+        /// The cause of the closure was not provided by the underlying platform.
+        /// </summary>
+        Undefined,
+
+        /// <summary>
+        /// The window itself was requested to close.
+        /// </summary>
+        WindowClosing,
+
+        /// <summary>
+        /// The window is closing due to a parent/owner window closing.
+        /// </summary>
+        OwnerWindowClosing,
+
+        /// <summary>
+        /// The window is closing due to the application shutting down.
+        /// </summary>
+        ApplicationShutdown,
+
+        /// <summary>
+        /// The window is closing due to the operating system shutting down.
+        /// </summary>
+        OSShutdown,
```

Comment:
Let's remove it for now

Reply:
My only reasoning for keeping it was: it's possible that someone has a check in their code such as `if (e.CloseReason == WindowCloseReason.ApplicationShutdown)` - if we then later add `OSShutdown` then that code is possibly no longer going to work correctly.

Having said that, I'm fine with removing it.